### PR TITLE
📖 Spelling

### PR DESCRIPTION
**Pull request by jsoref**
#### What kind of change does this PR introduce?

(Is it a bug fix, feature, docs update, something else?)

- [x] PR title follows the guidelines defined in our [pull request documentation](https://github.com/ossf/scorecard/blob/main/CONTRIBUTING.md#pr-process)

#### What is the current behavior?

The misspellings have been reported at 
https://github.com/jsoref/scorecard/actions/runs/7565320059/attempts/1#summary-20600890152

#### What is the new behavior (if this is a feature change)?**

- [ ] Tests for the changes have been added (for bug fixes/features)

The action will report that the changes in this PR would make it happy: 
https://github.com/jsoref/scorecard/actions/runs/7565320271/attempts/1#summary-20600890502

#### Special notes for your reviewer

I'm happy to drop any particular changes / changes to any particular files/paths or to split this as requested.

#### Does this PR introduce a user-facing change?

For user-facing changes, please add a concise, human-readable release note to
the `release-note`

(In particular, describe what changes users might need to make in their
application as a result of this pull request.)

* I'm concerned that the changes to `jsonBranchProtectionSettings` may be breaking. I think most of the other changes are harmless...

```release-note

```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -247,7 +247,7 @@ build-attestor-docker: ## Build scorecard-attestor Docker image
 build-attestor-docker:
 	DOCKER_BUILDKIT=1 docker build . --file attestor/Dockerfile \
 		--tag scorecard-attestor:latest \
-		--tag scorecard-atttestor:$(GIT_HASH)
+		--tag scorecard-attestor:$(GIT_HASH)
```
Comment (jsoref) [inline Makefile:250]: I'm uncertain about this... It seems quite strange...

File: README.md [hunk 2]
```diff
@@ -559,7 +559,7 @@ risk level.
 ### Report Problems
 
 If you have what looks like a bug, please use the
-[Github issue tracking system.](https://github.com/ossf/scorecard/issues) Before
+[GitHub issue tracking system.](https://github.com/ossf/scorecard/issues) Before
```
Comment (jsoref) [inline README.md:562]: brand

File: checks/evaluation/pinned_dependencies.go [hunk 3]
```diff
@@ -32,7 +32,7 @@ type pinnedResult struct {
 
 // Structure to host information about pinned github
 // or third party dependencies.
-type worklowPinningResult struct {
+type workflowPinningResult struct {
```
Comment (jsoref) [inline checks/evaluation/pinned_dependencies.go:35]: I believe that lowercase items are deemed private. I've avoided changing one or two items with capital names...

File: checks/fileparser/github_workflow_test.go [hunk 4]
```diff
@@ -103,7 +103,7 @@ func TestGitHubWorkflowShell(t *testing.T) {
 		},
 		{
 			name:           "shell specified in step",
-			filename:       "../testdata/.github/workflows/github-workflow-shells-speficied-step.yaml",
+			filename:       "../testdata/.github/workflows/github-workflow-shells-specified-step.yaml",
```
Comment (jsoref) [inline checks/fileparser/github_workflow_test.go:106]: this file is renamed...

File: checks/raw/binary_artifact.go [hunk 5]
```diff
@@ -278,7 +278,7 @@ func checkWorkflowValidatesGradleWrapper(path string, content []byte, args ...in
 	return true, nil
 }
 
-// fileExists checks if a file of name name exists, including within
+// fileExists checks if a file of name exists, including within
```
Comment (jsoref) [inline checks/raw/binary_artifact.go:281]: Actually, this is probably better as:
```suggestion
// fileExists checks if a file named `name` exists, including within
```

File: checks/raw/pinned_dependencies_test.go [hunk 6]
```diff
@@ -449,7 +449,7 @@ func TestFileIsInVendorDir(t *testing.T) {
 		},
 		{
 			name:     "misspelled vendor dir",
-			filename: "a/vendorr/Dockerfile",
+			filename: "a/vendor_/Dockerfile",
```
Comment (jsoref) [inline checks/raw/pinned_dependencies_test.go:452]: I understand that the goal of this test is to check for a misspelled item. I think that this misspelling should be functionally equivalent for validating the feature while making it easier to catch other misspellings in this project.

-- That said, I'm happy to drop this or anything else.

File: checks/raw/testdata/Dockerfile-gsutil-file [hunk 7]
```diff
@@ -21,11 +21,11 @@ RUN /tmp/exe
 RUN gsutil gs://exe /tmp/exe1 && /tmp/exe1
 RUN gsutil gs://exe /tmp/exe2 ; /tmp/exe2
 
-RUN gsutil gs://bucet/exe6 /path/to/
+RUN gsutil gs://bucket/exe6 /path/to/
```
Comment (jsoref) [inline checks/raw/testdata/Dockerfile-gsutil-file:24]: I'm assuming this doesn't care about the spelling...

File: checks/security_policy_test.go [hunk 8]
```diff
@@ -61,7 +61,7 @@ func TestSecurityPolicy(t *testing.T) {
 		},
 		{
 			name: "docs/security.md",
-			path: "./testdata/securitypolicy/04_textAndDisclosureVuls",
+			path: "./testdata/securitypolicy/04_textAndDisclosureVulns",
```
Comment (jsoref) [inline checks/security_policy_test.go:64]: these files are renamed below...

File: checks/testdata/.github/workflows/github-workflow-shells-specified-step.yaml
Comment (jsoref) [inline checks/testdata/.github/workflows/github-workflow-shells-specified-step.yaml:1]: this rename is for the first set of renames...

File: checks/testdata/binaryartifacts/workflows/verify-outdated-action.yaml [hunk 9]
```diff
@@ -2,7 +2,7 @@ name: "GW Validate Workflow"
 on: [push, pull_request]
 
 jobs:
-  gw_validat3:
+  gw_validate:
```
Comment (jsoref) [inline checks/testdata/binaryartifacts/workflows/verify-outdated-action.yaml:5]: ??

This change should be harmless...

File: checks/testdata/securitypolicy/04_textAndDisclosureVulns
Comment (jsoref) [inline checks/testdata/securitypolicy/04_textAndDisclosureVulns:1]: this pair of renames is for the second called out rename dependent item

File: cmd/package_managers_test.go [hunk 10]
```diff
@@ -336,7 +336,7 @@ func Test_fetchGitRepositoryFromPYPI(t *testing.T) {
     "description": "UNKNOWN",
     "description_content_type": null,
     "docs_url": null,
-    "downoad_url": null,
+    "download_url": null,
```
Comment (jsoref) [inline cmd/package_managers_test.go:339]: ??

File: pkg/json_raw_results.go [hunk 11]
```diff
@@ -63,7 +63,7 @@ type jsonBranchProtectionSettings struct {
 	RequiresLinearHistory               *bool    `json:"requiredLinearHistory"`
 	DismissesStaleReviews               *bool    `json:"dismissesStaleReviews"`
 	EnforcesAdmins                      *bool    `json:"enforcesAdmin"`
-	RequiresStatusChecks                *bool    `json:"requiresStatuChecks"`
+	RequiresStatusChecks                *bool    `json:"requiresStatusChecks"`
```
Comment (jsoref) [inline pkg/json_raw_results.go:66]: ⚠️  I'm worried that this is a breaking change. I'm happy to drop this if that's the case.

**Comment by pnacht** [comment]
Might it be better to split this into two PRs? One with the breaking changes and one without?

The team may prefer to leave the breaking changes as-is to avoid a major-version bump over these typos, but the non-breaking stuff should be easier to get merged.

**Comment by jsoref** [comment]
@pnacht: sure. Before I break things up, it'd be helpful if someone could confirm if that's the only one.

File: checks/raw/testdata/Dockerfile-gsutil-file (hunk 7 above)
Comment (spencerschrock) [reply checks/raw/testdata/Dockerfile-gsutil-file:24]: we'll see if the tests pass. if they do, should be fine.

If they dont, may need to update somewhere else.

File: cmd/package_managers_test.go (hunk 10 above)
Comment (spencerschrock) [reply cmd/package_managers_test.go:339]: if tests pass, fine with the change

File: pkg/json_raw_results.go (hunk 11 above)
Comment (spencerschrock) [reply pkg/json_raw_results.go:66]: didn't dig into it too much, but let's just drop it

**Review by spencerschrock — APPROVED** [review]

**Comment by jsoref** [comment]
Thanks @spencerschrock (this comment is mostly present so that it's easier to see the diff for the force push here in case there's another force push later).

File: checks/raw/testdata/Dockerfile-gsutil-file (hunk 7 above)
Comment (jsoref) [reply checks/raw/testdata/Dockerfile-gsutil-file:24]: All tests passed

File: cmd/package_managers_test.go (hunk 10 above)
Comment (jsoref) [reply cmd/package_managers_test.go:339]: All tests passed.

File: Makefile (hunk 1 above)
Comment (raghavkaul) [reply Makefile:250]: Should be fine, thanks for the fix!

**Comment by jsoref** [comment]
So, should this be 👻 ? 🐛 ? I think we're ruled out the things we think are ⚠️ worthy...
I don't think it's ✨ and while some of the changes are 📖, many aren't.

Is it :seedling: because it's "other"?

**Comment by spencerschrock** [comment]
> So, should this be 👻 ? 🐛 ? I think we're ruled out the things we think are ⚠️ worthy... I don't think it's ✨ and while some of the changes are 📖, many aren't.
> 
> Is it 🌱 because it's "other"?

I was debating between `:seedling:` and `:book:`, so happy to accept either. 
I usually merge typo fixes as `:book:` since they don't change any of the behavior, even if it's strictly not documentation.

**Comment by jsoref** [comment]
Sold. Note that it might be worthwhile to update the documentation to that effect (to some extent, projects should try to reduce "cognitive load" on external contributors).

Thanks for the fast iteration, I appreciate it.

**Pull request merged**